riot_api/methods/get_rank.py

```diff
@@ -0,0 +1,132 @@
+"""
+Data processing the data from riot API
+"""
+import pydash
+from db.db import Session
+from db.models.summoners import Summoners
+
+from utils.utils import get_file_path
+
+from .. import watcher, MY_REGION
+
+
+session = Session()
+
+# Get summoner rank.
+def get_summoner_rank(name: str):
+    """Gets the summoner's rank information from riot watcher api
+    Parameters:
+    name (str): name of the summoner
+
+    Returns:
+    summoner_profile (dict): rank information about the summoner
+
+    """
+    # TODO BEFORE REQUESTING DATA, CHECK DB AND UPDATE_TIME TO SEE IF WE ALREADY HAVE DATA.
+    # IF WE HAVE DATA BUT UPDATE_TIME IS OVER THE CONSTRAINT, WE SHOULD DELETE THE ROW.
+
+    # We need to get id
+    user = watcher.summoner.by_name(MY_REGION, name)
+
+    # First check if we have existing record for given summoner name
+    # Create query; TODO: check for update_time
+    summoner_cached_query = session.query(Summoners).filter(
+        Summoners.summoner_name == user["name"]
+    )
+
+    # Execute query
+    summoner_cached = summoner_cached_query.one_or_none()
```

Comment:
`cached` handling -> I think this wlill get used a lot. better to factor it out later.